### PR TITLE
feat(src): add filterIndexed

### PR DESCRIPTION
**Pull request by willblake01**
expose index to the predicate function

Ref #1346

# Pull request template

Please use the following [PR](https://github.com/char0n/ramda-adjunct/pull/905/files) as an example
of pull request. This PR deals with adding a new feature called `allSettledP`.

If you have any additional questions please don't hesitate to contact any of the core committers.

### CODE REVIEW

**Comment by willblake01** [comment]
@char0n This was a bit of a stretch for me. I added the test case from the issue but I'm unsure of how to add proper test coverage. Please feel free to add comments to point me in the right direction or assign to someone else if you don't have the time.

**Comment by char0n** [comment]
Good start. Let's move step by step here to make it fully compatible with ramda-adjunct codebase.

File: src/filterIndexed.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * @typedef Idx = Number
+ * @param {Function} fn The predicate function
+ * @param {Array} list the list to filter
+ * @param {*} val
+ * @return {Array} A filterable containing the value at a given index
+ * @see {@link http: //ramdajs.com/docs/#addIndex|R.addIndex}, {@link https://ramdajs.com/docs/#filter}
+ * @example
+ *
+ * filterIndexed((val, idx) => idx === 0, [1, 2, 3]); // => [1]
```
Comment (char0n) [inline src/filterIndexed.js]: Let's use example of original [R.filter](https://ramdajs.com/docs/#filter) function.
```suggestion
const isEven = n => n % 2 === 0;

R.filterIndexed((val, idx) => isEven(idx), [1, 2, 3, 4]); //=> [1, 3]
```

File: src/filterIndexed.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * @category List
+ * @typedef Idx = Number
+ * @sig Filterable f => ((a, Idx, f a) -> Boolean) -> f a -> f a
+ * @param {Function} fn The predicate function
+ * @param {Array} list The list to filter
+ * @return {Array} Filterable
+ * @see {@link http://ramdajs.com/docs/#addIndex|R.addIndex}, {@link http://ramdajs.com/docs/#filter|R.filter}
+ * @example
+ *
+const isEven = n => n % 2 === 0;
```
Comment (char0n) [inline src/filterIndexed.js]:
```suggestion
* const isEven = n => n % 2 === 0;
```

File: src/filterIndexed.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
```
Comment (char0n) [inline src/filterIndexed.js]:
```suggestion
 *
```

File: src/filterIndexed.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+R.filterIndexed((val, idx) => isEven(idx), [1, 2, 3, 4]); //=> [1, 3]
```
Comment (char0n) [inline src/filterIndexed.js]:
```suggestion
* R.filterIndexed((val, idx) => isEven(idx), [1, 2, 3, 4]); //=> [1, 3]
```

File: src/filterIndexed.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * @category List
+ * @typedef Idx = Number
+ * @sig Filterable f => ((a, Idx, f a) -> Boolean) -> f a -> f a
+ * @param {Function} fn The predicate function
+ * @param {Array} list The list to filter
+ * @return {Array} Filterable
+ * @see {@link http://ramdajs.com/docs/#addIndex|R.addIndex}, {@link http://ramdajs.com/docs/#filter|R.filter}
+ * @example
+ *
+* const isEven = n => n % 2 === 0;
```
Comment (char0n) [inline src/filterIndexed.js]:
```suggestion
 * const isEven = n => n % 2 === 0;
```

File: src/filterIndexed.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ *
+* R.filterIndexed((val, idx) => isEven(idx), [1, 2, 3, 4]); //=> [1, 3]
```
Comment (char0n) [inline src/filterIndexed.js]:
```suggestion
 * R.filterIndexed((val, idx) => isEven(idx), [1, 2, 3, 4]); //=> [1, 3]
```

**Comment by char0n** [comment]
Let's move forward. We're missing in this PR two additional files. Take a look at this model [PR](https://github.com/char0n/ramda-adjunct/pull/1338/files) which shows how to add your function type to `index.d.ts` and how to [test it](https://github.com/char0n/ramda-adjunct/pull/1338/files#diff-eb717d159c99dce1cf4df3804423eefb).

**Comment by char0n** [comment]
@willblake01 will you continue with this PR?

**Comment by willblake01** [comment]
@char0n I'm trying to learn some typescript and how to properly add test coverage. It's a bit of a stretch for me as I'm a novice coder but I'm enjoying trying to figure it out. Feel free to assign to someone else if you need it finished.

**Comment by char0n** [comment]
@willblake01 no worries. Closing this for now. Anybody interested can pick up where you left off. THank you!

**Pull request closed without merging**